PostHog/PostHogSDK.swift

```diff
@@ -723,4 +733,12 @@ let maxRetryDelay = 30.0
         }
         capture("Application Backgrounded")
     }
+
+    private func sanitizeDicionary(_ dict: [String: Any]?) -> [String: Any]? {
+        if dict == nil || dict!.isEmpty || !JSONSerialization.isValidJSONObject(dict!) {
+            return nil
+        }
+
+        return dict
+    }
```

Comment:
1. If it isn't serializable, we probably want to hedgeLog something about that for debugging purposes ("why are my events missing properties??")
2. We could do this on a per-property basis, filtering out any properties that are not json serializable. This would allow some properties to get through fine (`$screen_name` for example) even if one property was not serializable.
3. If we don't do 2. then I think we need to actually drop the whole event. Otherwise, it is likely events will come through with completely meaningless properties which feels worse than coming through with only one property missing
4. We could even add a special property like `$serialization_error_for_keys: ['my_bad_property']` so that we can indicate in the UI that something went wrong

Reply:
1. good point, will add logging.
2. that won't happen since I only call `sanitizeDicionary` for the given properties, not the ones generated by the SDK.
eg:

```
let props = [
            "$screen_name": screenTitle,
        ].merging(sanitizeDicionary(properties) ?? [:]) { prop, _ in prop }
```

3. I agree, there's just no easy way to check if every property is serializable or not in an easy way.
The only easy solution I found is to wrap a property in a list or dictionary and call `JSONSerialization.isValidJSONObject` but then there's a huge overhead, besides the need to be recursive.
I am open to ideas, all the other ideas compatible with ObjC and Swift would be tons of code or 3rd party libs, See `Any background context you want to provide?` section.

4. That would be awesome, but again, see the item 3.

Reply:
For context, iOS SDK v2 does not do it either, just document via an annotation https://github.com/PostHog/posthog-ios/blob/8500ffd930fb53670f6ad8c10f43d2efe3321089/PostHog/Classes/PHGSerializableValue.h#L13 + that https://github.com/PostHog/posthog-ios/blob/8500ffd930fb53670f6ad8c10f43d2efe3321089/PostHog/Internal/PHGPostHogUtils.m#L107 which is very slow for nested properties.

Reply:
I don't think we need to do it nested, just top level properties would be fine and then I don't think the overhead should be that much (but happy to be proved wrong) to iterate 10 properties?

Reply:
@benjackwhite If one of the nested properties is not serializable, the whole JSONSerialization fails anyway, so we have to do it, assuming that all Dicts are quite small, it should be fine, let me see if I can get something quickly without a huge amount of LoC that works for ObjC as well

Reply:
@benjackwhite pushed some changes, I still only do for the given properties since the properties set by the SDK are always serializable, so this spares a bit of the overhead.
Also, only for high-level properties, if a nested dict or array isn't serializable, we drop it and log it out.
I also added the `URL` and `Date` parsing to be compatible with the older version.